src/BlazorWebView/src/Maui/Windows/BlazorWebView.Windows.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+
+namespace Microsoft.AspNetCore.Components.WebView.Maui
+{
+	public partial class BlazorWebView
+	{
+		/// <inheritdoc/>
+		protected override void OnPropertyChanging(string? propertyName = null)
+		{
+			if (propertyName == nameof(Window) && Window is not null)
+				Window.Destroying -= Window_Destroying;
+		}
+
+		/// <inheritdoc/>
+		protected override void OnPropertyChanged(string? propertyName = null)
+		{
+			if (propertyName == nameof(Window) && Window is not null)
+				Window.Destroying += Window_Destroying;
+		}
+
+		private void Window_Destroying(object? sender, EventArgs e)
+		{
+			// see: https://github.com/microsoft/microsoft-ui-xaml/issues/6872
+			((BlazorWebViewHandler)Handler!).PlatformView.Close();
```

Comment:
> Are all of the `.`'s here actually guaranteed to be non-null? Or should we play it safe and do `?.` for everything?

A similar assumption is made [here](https://github.com/dotnet/maui/blob/d116b03456ee059e76ccc023ecee80b14a3af3e1/src/BlazorWebView/src/Maui/BlazorWebView.cs#L61), so I think it should be fine? I guess technically since `Handler` is publicly settable, someone could mess with it and cause a null ref here. If I change it here, should I change it in the other location?

> Do we need the down-cast?

Yes, since the type of `Handler` is the non-generic `IViewHandler`.

Reply:
Re: The `.`'s: the reason I'd prefer to be extra safe here is that in other parts of the control's code we tend to "know what's going on" in terms of the order of things happening. My concern is that in this case the window changing might happen somewhat unexpectedly, because it's often user-controlled (for example, pressing the `X` button in the UI whenever they want), so I wonder if there are cases where the window could be destroyed while we're still building things up, and so handlers/views/whatevers aren't in a stable state. I have no evidence for this, but I have a feeling in my gut.

Re: The downcast: Sad 😢. I will live with this.

Reply:
I've removed the `!` in favor of a `?.`, but I'm leaving it as `PlatformView.Close()` rather than `PlatformView?.Close()` because the `PlatformView` property throws an `InvalidOperationException` anyway if the backing property is `null` (so the `?.` would do nothing in that case).